meta-sokol-flex-distro/conf/distro/sokol-flex.conf

```diff
@@ -165,6 +151,39 @@ TOOLCHAIN_TARGET_TASK:append = " ${TOOLCHAIN_TARGET_TASK_EXTRA}"
 SSTATE_MIRROR_SITES ?= ""
 SSTATE_MIRRORS += "${@''.join(r'file://.* %s/PATH;downloadfilename=PATH \n' % u for u in '${SSTATE_MIRROR_SITES}'.split())}"
 ## }}}1
+## BitBake Configuration and Behavior {{{1
+# Disk Space Monitoring during the build
+#
+# Monitor the disk space during the build. If there is less that 1GB of space or less
+# than 100K inodes in any key build location (TMPDIR, DL_DIR, SSTATE_DIR), gracefully
+# shutdown the build. If there is less than 100MB or 1K inodes, perform a hard halt
+# of the build. The reason for this is that running completely out of space can corrupt
+# files and damages the build in ways which may not be easily recoverable.
+# It's necessary to monitor /tmp, if there is no space left the build will fail
+# with very exotic errors.
+BB_DISKMON_DIRS ??= "\
+    STOPTASKS,${TMPDIR},1G,100K \
+    STOPTASKS,${DL_DIR},1G,100K \
+    STOPTASKS,${SSTATE_DIR},1G,100K \
+    STOPTASKS,/tmp,100M,100K \
+    HALT,${TMPDIR},100M,1K \
+    HALT,${DL_DIR},100M,1K \
+    HALT,${SSTATE_DIR},100M,1K \
+    HALT,/tmp,10M,1K"
+
+# Support shallow git tarballs
+BB_GENERATE_SHALLOW_TARBALLS ??= "1"
+BB_GIT_SHALLOW ??= "1"
+BB_GIT_SHALLOW_DEPTH_DEFAULT ??= "1"
+BB_GIT_SHALLOW_DEPTH ??= "${@'${BB_GIT_SHALLOW_DEPTH_DEFAULT}' if not '${BB_GIT_SHALLOW_REVS}' else '0'}"
+BB_GIT_SHALLOW_REVS ??= ""
+
+# For our kernel repository, we want to ship the user the history which they
+# can't easily re-acquire from upstream.
+LINUX_VERSION_TAG ??= "v${@'.'.join('${LINUX_VERSION}'.split('.')[:2])}"
+LINUX_VERSION ??= ""
+BB_GIT_SHALLOW_REVS:pn-linux-mel ??= "${@'${LINUX_VERSION_TAG}' if '${LINUX_VERSION_TAG}' and '${LINUX_VERSION}' else ''}"
```

Comment:
Shouldn't this be done through the BSP layer?

Reply:
I've opened PRs on this repo and the meta-flex-ref-bsps repo to get this moved.